tests/functional/lib/helpers.js

```diff
@@ -402,40 +414,39 @@ define([
       });
   }
 
-  function openFxaFromRp(context, page, urlSuffix, untrusted) {
-    var app = untrusted ? UNTRUSTED_OAUTH_APP : OAUTH_APP;
-
-    var additionalQueryParams;
-    if (urlSuffix) {
-      additionalQueryParams = Querystring.parse(urlSuffix.replace(/^\?/, ''));
-    }
+  function openFxaFromRp(context, page, options) {
+    options = options || {};
+    var app = options.untrusted ? UNTRUSTED_OAUTH_APP : OAUTH_APP;
+    var expectedHeader = options.header || '#fxa-' + page.replace('_', '-') + '-header';
+    var queryParams = options.query || {};
 
     // force_auth does not have a button on 123done, instead this is
     // only available programatically. Load the force_auth page
     // with only the email initially, then reload with the full passed
     // in urlSuffix so things like the webChannelId are correctly passed.
+
     if (page === 'force_auth') {
-      var email = additionalQueryParams.email;
+      var email = queryParams.email;
       var emailSearchString = '?' + Querystring.stringify({ email: email });
-      return openPage(context, app + 'api/force_auth' + emailSearchString, '#fxa-force-auth-header')
+      var endpoint = app + 'api/force_auth' + emailSearchString;
+      return openPage(context, endpoint, expectedHeader)
         .then(function () {
-          return reOpenWithAdditionalQueryParams(context, additionalQueryParams, '#fxa-force-auth-header');
+          if (Object.keys(queryParams).length > 1) {
+            return reOpenWithAdditionalQueryParams(context, queryParams, expectedHeader);
+          }
         });
     }
 
     return openPage(context, app, '.ready #splash .' + page)
-      .findByCssSelector('.ready #splash .' + page)
-        .click()
-      .end()
+      .then(click('.ready #splash .' + page))
 
       // wait until the page fully loads or else the re-load with
       // the suffix will blow its lid when run against latest.
-      .findByCssSelector('#fxa-' + page + '-header')
-      .end()
+      .then(testElementExists(expectedHeader))
 
       .then(function () {
-        if (additionalQueryParams) {
-          return reOpenWithAdditionalQueryParams(context, additionalQueryParams, '#fxa-' + page + '-header');
+        if (Object.keys(queryParams)) {
```

Comment:
...and actually, `Object.keys(undefined)` throws anyway, so it will never evaluate to `false` regardless of what `queryParams` is. Did you mean to check `.length`?